### PR TITLE
Paypal in lesson header test

### PR DESCRIPTION
**Pull request by drjwbaker**
Testing paypal in lesson header per https://github.com/programminghistorian/jekyll/issues/1503

### CODE REVIEW

**Comment by jenniferisasi** [comment]
How about a little coin in place of the signs? Like the little person and the little dock on the other two banners :) 

Looks like the other icons are from https://fontawesome.com/icons/lock (pulled it from the file code) so i tracked this icon: https://fontawesome.com/icons/coins?style=solid  
I also like the piggy-bank

**Comment by walshbr** [comment]
I'll implement one of these in a second and ping you both when i have something on the branch.

**Comment by walshbr** [comment]
Sorry this was taking forever to regenerate locally, which was making things slow. Check it now! The coin icon is not available in the version of font awesome we're using, and upgrading means we lose the old icons. So I went with the credit card. I also changed the text to "Support PH" because "Support the Programming Historian" is so many more characters than "Peer Reviewed" and "CC by 4.0" that there weren't going to be any good design solutions for it. Does that text work for you all? Could also go with with "Support us" or "Donate" @drjwbaker @jenniferisasi

**Comment by jenniferisasi** [comment]
I like it! I would change Support PH to Support us, but it is not a big deal.

**Comment by walshbr** [comment]
I'm fine with either one really, since I drew the line at character counts :) Do you have one you would prefer @drjwbaker? I note that in #1503 there was a question of ambiguity over who was being supported.

**Comment by drjwbaker** [comment]
Yeah, 'Support Us' could mean the authors in this context.

**Comment by jenniferisasi** [comment]
> Yeah, 'Support Us' could mean the authors in this context.

Ah, then Support PH it is.

**Comment by walshbr** [comment]
In that case this should be good and just in need of translation, right @drjwbaker?

**Comment by drjwbaker** [comment]
Thanks @walshbr & @jenniferisasi. Pinging @acrymble as I know you care about such things. Does this look good to you? If so, we'll push towards translation.

**Comment by drjwbaker** [comment]
> In that case this should be good and just in need of translation, right @drjwbaker?

@acrymble appears to be happy, so yes just translation needed. 

@programminghistorian/spanish-team @rivaquiroga: see https://deploy-preview-1520--ph-dev.netlify.com/en/lessons/analyzing-documents-with-tfidf. Only translation needed is the phrase 'Support PH'. Is 'Apoyar PH' correct?

@programminghistorian/french-team @spapastamkou: see https://deploy-preview-1520--ph-dev.netlify.com/en/lessons/analyzing-documents-with-tfidf. Only translation needed is the phrase 'Support PH'. Is 'Soutenir PH' correct?

**Comment by mdlincoln** [comment]
So this is what I'm seeing on the preview deploy:

Screen Shot 2019-11-04 at 8 57 26 AM

Are you sure the patreon script link is correctly added in?

File: _includes/footer.html [hunk 1]
```diff
@@ -8,6 +8,16 @@
...
 </div>
+ 
+<div class="d-flex flex-wrap justify-content-center footer-head">
+ <form action="https://www.paypal.com/cgi-bin/webscr" method="post" target="_top">
+  <input type="hidden" name="cmd" value="_s-xclick" />
+  <input type="hidden" name="hosted_button_id" value="7BGHUZRVS4LYL" />
+  <input type="image" src="https://www.paypalobjects.com/en_GB/i/btn/btn_donate_LG.gif" border="0" name="submit" title="PayPal - The safer, easier way to pay online!" alt="Donate with PayPal button" />
+  <img alt="" border="0" src="https://www.paypal.com/en_GB/i/scr/pixel.gif" width="1" height="1" />
+</form>
+<a href="https://www.patreon.com/bePatron?u=11816614" data-patreon-widget-type="become-patron-button">Become a Patron!</a><script async src="https://c6.patreon.com/becomePatronButton.bundle.js"></script>
```
Comment (mdlincoln) [inline _includes/footer.html]: confirm that this is all that's needed to display Patron button

**Review by mdlincoln — CHANGES_REQUESTED** [review]

File: _includes/footer.html (hunk 1 above)
Comment (drjwbaker) [reply _includes/footer.html]: apologies. the footer buttons got left in after my initial playing. If we are going to keep them in (and no need not to) by saying "confirm that this is all that's needed to display Patron button" do you mean the changes you've made work?

Comment (mdlincoln) [reply _includes/footer.html]: I have made no changes...?

I do not like the footer buttons - I think the lesson buttons are fine. If you intend to remove the footer buttons, then do so.

**Comment by fdlaramee** [comment]
@drjwbaker Yes, "Soutenir PH" is appropriate. Thanks!

**Comment by drjwbaker** [comment]
@walshbr I think I'll added the snippet thingy correctly..

File: _includes/footer.html (hunk 1 above)
Comment (drjwbaker) [reply _includes/footer.html]: Sorry. I didn't understand your first message. I'll remove the footers. Aren't my favourite either.

**Comment by walshbr** [comment]
Yep that's correct use of the snippets @drjwbaker 🎉  That portion looks good to me in the preview.

**Review by mdlincoln — CHANGES_REQUESTED** [review]
Right now, the links only point to the English "support us" page, no matter which language page you're coming from. We should have the links point dynamically to the correct language page by relying on the existing snippet here: https://github.com/programminghistorian/jekyll/blob/gh-pages/_data/snippets.yml#L143

the `href` part of the link would need to be updated to read like `https://programminghistorian.org/{{ page.lang }}/{{ site.data.snippets.menu-contribute-support[page.lang].link}}`

**Comment by drjwbaker** [comment]
Good catch @mdlincoln. I think I've done it right. Though I'm not sure if (if at all?) spaces between curly brackets do anything.

**Comment by drjwbaker** [comment]
> Yep that's correct use of the snippets @drjwbaker 🎉 That portion looks good to me in the preview.

Shock horror.

**Comment by drjwbaker** [comment]
Did it wrong. Let me try again..

**Review by mdlincoln — APPROVED** [review]

**Comment by drjwbaker** [comment]
Sorry. Me again. How do we use the snippets to make the link go to the correct language sub-section of a page, rather than just the correct language page? So https://programminghistorian.org/es/apoyanos#donaciones rather than https://programminghistorian.org/es/apoyanos for https://deploy-preview-1520--ph-dev.netlify.com/es/lecciones/analisis-temporal-red.

**Comment by drjwbaker** [comment]
pinging @programminghistorian/technical-team

**Comment by walshbr** [comment]
Hmm…I think you would need to add a new snippet for the subsection you're interested in so the links would become something like this -

https://programminghistorian.org/{{ page.lang }}/{{ site.data.snippets.menu-contribute-support[page.lang].link}}#{{ site.data.snippets.menu-contribute-support[page.lang].subsection }}

Though note that the trailing piece after the slash would be - the actual structure following the slash would depend on what you name the snippet / where it is / etc.. You'd need to have some way to store the translated name of the subheader in a snippet so that you can call it in the link by referencing the page language.
